### PR TITLE
Issue #7690: ClassDataAbstractionCoupling doc update

### LINKED ISSUE #7690: update doc for ClassDataAbstractionCoupling
https://checkstyle.org/config_metrics.html#ClassDataAbstractionCoupling

Add example for each existing config to show what code is ok and what code cause violation.
Good example is https://checkstyle.org/config_naming.html#AbstractClassName
Please add examples for all properties of Check.

look at referenced commits of #5832, to see what and how should be changed in code.

please play with module/Check by means of https://checkstyle.org/cmdline.html to understand how it works.

------------

PLEASE READ https://github.com/checkstyle/checkstyle/wiki/Good-practice-on-how-to-update-examples

### PR DESCRIPTION
**Pull request by shashwat70**
Closes #7690  : update doc for ClassDataAbstractionCoupling

Site: https://shashwat70.github.io/site/config_metrics.html#ClassDataAbstractionCoupling

Default config (Good):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="ClassDataAbstractionCoupling"/>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    Place place = new Place(); // Counted, 3
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

Default config (Violating):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
      <module name="ClassDataAbstractionCoupling"/>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    A1 a1=new A1();
    A2 a2=new A2();
    A3 a3=new A3();
    A4 a4=new A4();
    A5 a5=new A5();
    Place place = new Place(); // violation, total is 8
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:1:1: Class Data Abstraction Coupling is 8 (max allowed is 7) classes [A1, A2, A3, A4, A5, Date, Place, Time]. [ClassDataAbstractionCoupling]
Audit done.
Checkstyle ends with 1 errors.
```

max property config (Good):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="max" value="2"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
  }

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

max property config (Violating):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="max" value="2"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    Place place = new Place(); // violation, total is 3
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:1:1: Class Data Abstraction Coupling is 3 (max allowed is 2) classes [Date, Place, Time]. [ClassDataAbstractionCoupling]
Audit done.
Checkstyle ends with 1 errors.
```

excludedClasses property config (Good):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludedClasses" value="HashMap, HashSet, Place"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // Ignored
    Map map = new HashMap(); // Ignored
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    A1 a1=new A1();
    A2 a2=new A2();
    A3 a3=new A3();
    A4 a4=new A4();
    A5 a5=new A5();
    Place place = new Place(); // Ignored
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

excludedClasses property config (Violating):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludedClasses" value="HashMap, HashSet, Place"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // Ignored
    Map map = new HashMap(); // Ignored
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    A1 a1=new A1();
    A2 a2=new A2();
    A3 a3=new A3();
    A4 a4=new A4();
    A5 a5=new A5();
    Space space = new Space(); // violation, total is 8
  }

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:1:1: Class Data Abstraction Coupling is 8 (max allowed is 7) classes [A1, A2, A3, A4, A5, Date, Space, Time]. [ClassDataAbstractionCoupling]
Audit done.
Checkstyle ends with 1 errors.
```

excludedClassesRegexps property config (Good):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludeClassesRegexps" value=".*Reader$"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    A1 a1=new A1();
    A2 a2=new A2();
    A3 a3=new A3();
    A4 a4=new A4();
    A5 a5=new A5();
    BufferedReader br = new BufferedReader(); // Ignored
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

excludedClassesRegexps property config (Violating):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludeClassesRegexps" value=".*Reader$"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
class InputClassCoupling {
    Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
    Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
    Date date = new Date(); // Counted, 1
    Time time = new Time(); // Counted, 2
    A1 a1=new A1();
    A2 a2=new A2();
    A3 a3=new A3();
    A4 a4=new A4();
    A5 a5=new A5();
    File file = new File(); // violation, total is 8
  }
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:1:1: Class Data Abstraction Coupling is 8 (max allowed is 7) classes [A1, A2, A3, A4, A5, Date, File, Time]. [ClassDataAbstractionCoupling]
Audit done.
Checkstyle ends with 1 errors.
```

excludedPackages property config (Good):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludedPackages" value="java.io"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
import java.io.BufferedReader;

class InputClassCoupling {
  Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
  Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
  Date date = new Date(); // Counted, 1
  Time time = new Time(); // Counted, 2
  A1 a1=new A1();
  A2 a2=new A2();
  A3 a3=new A3();
  A4 a4=new A4();
  A5 a5=new A5();
  BufferedReader br = new BufferedReader(); // Ignored
}

C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

excludedPackages property config (Violating):
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludedPackages" value="java.io"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
import java.util.StringTokenizer;

class InputClassCoupling {
  Set set = new HashSet(); // HashSet ignored due to default excludedClasses property
  Map map = new HashMap(); // HashMap ignored due to default excludedClasses property
  Date date = new Date(); // Counted, 1
  Time time = new Time(); // Counted, 2
  A1 a1=new A1();
  A2 a2=new A2();
  A3 a3=new A3();
  A4 a4=new A4();
  A5 a5=new A5();
  StringTokenizer st = new StringTokenizer(); // violation, total is 8
}
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
[ERROR] C:\Users\Shashwat\Documents\Test.java:3:1: Class Data Abstraction Coupling is 8 (max allowed is 7) classes [A1, A2, A3, A4, A5, Date, StringTokenizer, Time]. [ClassDataAbstractionCoupling]
Audit done.
Checkstyle ends with 1 errors.
```

Other explanatory example:
```
C:\Users\Shashwat\Documents>cat config.xml
<?xml version="1.0"?>
<!DOCTYPE module PUBLIC
          "-//Puppy Crawl//DTD Check Configuration 1.3//EN"
          "https://checkstyle.org/dtds/configuration_1_3.dtd">
<module name="Checker">
  <module name="TreeWalker">
    <module name="ClassDataAbstractionCoupling">
      <property name="excludedPackages" value="a.b"/>
    </module>
  </module>
</module>
C:\Users\Shashwat\Documents>cat Test.java
package a.b;

import a.b.Bar;
import a.b.c.Baz;

class Foo {
  Bar bar; // Will be ignored, located inside ignored a.b package
  Baz baz; // Will not be ignored, located inside a.b.c package
  Data data; // Will not be ignored, same file

  class Data {
    Foo foo; // Will not be ignored, same file
  }
}
C:\Users\Shashwat\Documents>java -jar checkstyle-8.29-all.jar -c config.xml Test.java
Starting audit...
Audit done.
```

### CODE REVIEW

**Comment by romani** [comment]
Please remove example of Max=5. Following example for Max=2 covers it.
Please place example under each config.

**Comment by rdiachenko** [comment]
@shashwat70 could you please reply 'done' to each review note, otherwise it is not clear what's the status and what was done.

**Comment by shashwat70** [comment]
Why did this check fail? I could'n 

> @shashwat70 could you please reply 'done' to each review note, otherwise it is not clear what's the status and what was done.

Yes sir, I will. Could you please review this https://github.com/checkstyle/checkstyle/pull/7906 first. These both are similar and I've done all changes in that one. I'll proceed with this one in a similar fashion once you confirm that what I did in the other one is right. @rdiachenko

**Comment by rdiachenko** [comment]
> > @shashwat70 could you please reply 'done' to each review note, otherwise it is not clear what's the status and what was done.
> 
> Yes sir, I will. Could you please review this #7906 first. These both are similar and I've done all changes in that one. I'll proceed with this one in a similar fashion once you confirm that what I did in the other one is right. @rdiachenko

@shashwat70 could you please update this PR with the knowledge you got from #7906 ? Please, don't forget to address Roman's comment - https://github.com/checkstyle/checkstyle/pull/7933#issuecomment-602812743

**Comment by shashwat70** [comment]
> > > @shashwat70 could you please reply 'done' to each review note, otherwise it is not clear what's the status and what was done.
> > 
> > 
> > Yes sir, I will. Could you please review this #7906 first. These both are similar and I've done all changes in that one. I'll proceed with this one in a similar fashion once you confirm that what I did in the other one is right. @rdiachenko
> 
> @shashwat70 could you please update this PR with the knowledge you got from #7906 ? Please, don't forget to address Roman's comment - [#7933 (comment)](https://github.com/checkstyle/checkstyle/pull/7933#issuecomment-602812743)

Done. @rdiachenko

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 1]
```diff
@@ -107,86 +107,189 @@
  * &lt;module name="ClassDataAbstractionCoupling"/&gt;
  * </pre>
  * <p>
- * To configure the check with a threshold of 5:
+ * Code examples:
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:110]: Please change to `Example:`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 2]
```diff
@@ -107,86 +107,189 @@
...
+ * </p>
+ * <p>
+ * The check passes without violations in the following.
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:113]: This is redundant, please remove.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 3]
```diff
@@ -107,86 +107,189 @@
...
- * &lt;module name="ClassDataAbstractionCoupling"&gt;
- *   &lt;property name="max" value="5"/&gt;
- * &lt;/module&gt;
+ * class InputClassCoupling {
+ *   Set set = new HashSet(); // Ignored due to default excludedClasses property
+ *   Map map = new HashMap(); // Ignored due to default excludedClasses property
+ *   Date date = new Date(); // Counted, 1
+ *   Time time = new Time(); // Counted, 2
+ *   Place place = new Place(); // Counted, 3
+ * }
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:122]: This example should contain violation for default configuration. Please update it accordingly.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 4]
```diff
@@ -107,86 +107,189 @@
...
  * </pre>
  * <p>
- * To configure the check with two excluded classes {@code HashMap} and {@code HashSet}:
+ * The check results in a violation in the following.
  * </p>
  * <pre>
- * &lt;module name="ClassDataAbstractionCoupling"&gt;
- *   &lt;property name="excludedClasses" value="HashMap, HashSet"/&gt;
- * &lt;/module&gt;
+ * class InputClassCoupling {
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:128]: Please remove and update the above example. There should be a single example for default configuration with a violation for default values.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 5]
```diff
@@ -107,86 +107,189 @@
...
  * &lt;module name="ClassDataAbstractionCoupling"&gt;
- *   &lt;property name="excludeClassesRegexps"
- *     value="^Array.*, .*Exception$"/&gt;
+ *   &lt;property name="max" value="2"/&gt;
  * &lt;/module&gt;
  * </pre>
  * <p>
- * The following example demonstrates usage of <b>excludedClasses</b> and
- * <b>excludeClassesRegexps</b> properties
+ * Code examples:
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:147]: `Code examples:` -> `Example:`

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 6]
```diff
@@ -107,86 +107,189 @@
...
  * <p>
- * Expected result is one class {@code Date}
+ * The check passes without violations in the following.
  * </p>
  * <pre>
- * &lt;module name="ClassDataAbstractionCoupling"&gt;
- *   &lt;property name="excludedClasses" value="ArrayList"/&gt;
- *   &lt;property name="excludeClassesRegexps" value="^Hash.*"/&gt;
- * &lt;/module&gt;
+ * class InputClassCoupling {
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:153]: Please make one code example for each config. Review and update other examples in the same way.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 7]
```diff
@@ -216,8 +319,8 @@
  * you should use direct import (e.g. {@code import java.math.BigDecimal}).
  * </p>
  * <p>
- * Also note, that checkstyle will not exclude classes within the same file
- * even if it was listed in the {@code excludedPackages} parameter.
+ * Also note, that checkstyle will not exclude classes within the same file even
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:322]: Looks like code formatting. Please revert.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java [hunk 8]
```diff
@@ -234,30 +337,22 @@
...
  * import a.b.c.Baz;
  *
- * public class Foo {
- *   public Bar bar; // Will be ignored, located inside ignored a.b package
- *   public Baz baz; // Will not be ignored, located inside a.b.c package
- *   public Data data; // Will not be ignored, same file
+ * class Foo {
+ *   Bar bar; // Will be ignored, located inside ignored a.b package
+ *   Baz baz; // Will not be ignored, located inside a.b.c package
+ *   Data data; // Will not be ignored, same file
```
Comment (rdiachenko) [inline src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:343]: What is the reason of this refactoring? Please revert if not needed.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 4 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:128]: https://github.com/checkstyle/checkstyle/blob/a231f5202c6769f9c664eb53a1bb370ec8788c96/src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassFanOutComplexityCheck.java#L97
The default config had 2 examples in this, one with and one without violations.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 6 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:153]: Same confusion.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 8 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:343]: https://github.com/checkstyle/checkstyle/pull/7906#discussion_r399785305
Please clarify.

**Comment by shashwat70** [comment]
Now I'm confused. My other PR for the similar update got merged recently. https://github.com/checkstyle/checkstyle/pull/7906
I've commented the basis of my confusions as a reply to each conversation above. Please clarify once and I'll be more than happy to make the required changes. @rdiachenko 
I'm sorry for my momentary non-compliance. I just wish seamlessness and uniformity to be preserved across checks.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 8 above)
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:343]: Ok, thanks for clarifying

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 4 above)
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:128]: Good point. Thanks for the link. Let is keep consistent.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 6 above)
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:153]: Thanks for the link. Let is keep consistent.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 3 above)
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:122]: No need anymore. There is a violation case below.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 2 above)
Comment (rdiachenko) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java:113]: No need, keeping consistent with ClassFanOutComplexityCheck

**Comment by rdiachenko** [comment]
@shashwat70 I resolved few notes. As you correctly pointed, let's keep consistency between already merged changes. Thanks for that.

Please review and fix left notes.

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 1 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 5 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java]: Done

File: src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java (hunk 7 above)
Comment (shashwat70) [reply src/main/java/com/puppycrawl/tools/checkstyle/checks/metrics/ClassDataAbstractionCouplingCheck.java]: Done

**Comment by shashwat70** [comment]
> Please review and fix left notes.

Aligned with ClassFanOutComplexityCheck. Please review. @rdiachenko

**Comment by shashwat70** [comment]
@rnveach ping

**Comment by shashwat70** [comment]
@rnveach ping

**Comment by romani** [comment]
@shashwat70 , sorry for delay, rnveach is over-occupied currently, I will try to take his PR on me.

**Pull request merged**